api/client/src/main/java/org/projectnessie/client/rest/ResponseCheckFilter.java

```diff
@@ -132,4 +139,44 @@ private static NessieError decodeErrorObject(
     }
     return error;
   }
+
+  /**
+   * Captures the first 2kB of the input in case the response is not parsable, to provide a better
```

Comment:
Should we skip this if the media type is binary, e.g. `application/octet-stream`?

Reply:
We can get garbage even if the content type is set to `text`... I guess the concern is about presenting that data to users... How about we exclude non-printable chars?

Reply:
What about printing a hex dump? This person has a nice solution: https://gist.github.com/jen20/906db194bd97c14d91df

Reply:
I doubt that is necessary. No HTTP server or gateway sends non-printable data for the kinds of errors we want to catch here.

Reply:
I think @dimas-b is rather worried by giant HTML spaghettis in the response... I've seen those once or twice. Presenting the data as a hex dump or stg in the style would avoid messing up with the logs.

Reply:
... and become unreadable for humanes

Reply:
b/c users report these kinds of errors when using tools

Reply:
I agree that the risk of getting binary output here is very low. As for HTML, hex dumping will make it harder to read indeed... so I'm ok with including the text as a simple string.